src/views/components/response_components/select_ta.vue

```diff
@@ -0,0 +1,150 @@
+                            return response.json();
+                        })
+                        .then(data => {
+                            this.ta_package = this.process_tas(data);
+                            this.all_terms = [201709, 201801, 201701];
+                            if (!this.course) {
+                                this.all_courses = Object.keys(this.ta_package);
+                            }
+                        })
+                        .catch(err => {
+                            this.$emit("error", err.toString());
+                        });
+                    break;
+                case "ta":
+                    this.target_ta = user_id;
+                    this.select_ta(
+                        user_id,
+                        this.user_name,
+                        term,
+                        course,
+                        user_id
+                    );
+                    break;
+                default:
+                    console.warn(
+                        "`user_type` of ",
+                        user_type,
+                        "is unrecognized."
+                    );
+            }
+        },
+        select_ta: function(ta_id, ta_name, term, course, user_id) {
+            this.$router.push({
+                name: "response_page",
+                query: {
+                    target_ta: ta_id,
+                    term: term,
+                    course: course,
+                    user_id: user_id,
+                    selected_ta: ta_name
+                }
+            });
+        },
+        // group ta by courses
+        // Input: {DATA: [{course_code: "CSC100", name: "Daniel", user_id: "ta1"},
+        //               {course_code: "CSC100", name: "Darren", user_id: "ta2"},
+        //               {course_code: "CSC101", name: "Daniel", user_id: "ta1"},
+        //               {course_code: "CSC101", name: "Darren", user_id: "ta2"}
+        //  ]}
+        // Output : { CSC100: [{name: "Daniel", id: "ta1"},{name: "Dareen", id: "ta2"}]
+        //            CSC101: [{name: "Daniel", id: "ta1"},{name: "Dareen", id: "ta2"}]
+        //}
+        process_tas: function(data) {
+            let ta_package = {};
+            for (let row of data.DATA) {
+                ta_package[row.course_code] = ta_package[row.course_code] || [];
+                ta_package[row.course_code].push({
+                    name: row.name,
+                    id: row.user_id
+                });
+            }
+            return ta_package;
+        },
+        // return false if object is null or empty
+        isEmpty: function(object) {
+            return !object || Object.keys(object).length === 0;
+        }
+    },
+    computed: {
+        //returns the updated component of term and course combined
+        // added date to be unique every time
+        term_and_course() {
+            return this.term, this.course, Date.now();
```

Comment:
Don't use the comma operator. Are you trying to return an array?

Reply:
the computed property changes when either term or course is changing(from the select element), this just returns a unique value. Then the watch property sees a change in the computed value and updates the page based on new specifications of term and course.

Reply:
I can change it to an object containing term and course

Reply:
IF you would like to call `this.term` and `this.course`, do

```
this.term;
this.course;
return Date.now();
```